### ISSUE TITLE
[Tune] suggest - resolving values too many times

### ISSUE THREAD

**Issue by RoeeEnbar** [ray-project/ray#6864]
suggest/variant_generator.py::_resolve_lambda_vars() resolves values more than once.

Specifically, on the first iteration the base values are resolved.
On the second iteration the conditional values are resolved with the former base values, but the base values are resolved again.
This results in conditional values and the values they were conditioned upon not having the same values.

Example:
```
    def test_basic(self):
        class MockTrainer(Trainable):
            def _train(self):
                return dict(mean_loss=1)

            def _stop(self):
                pass

        grid_size = 3
        ray.init(num_cpus=grid_size)

        config = {"a": {"eval": "random.uniform(0, 1)"},
                       "b": {"eval": "spec.config.a"}}

        analysis = tune.run(MockTrainer,
                            name='test_basic',
                            resources_per_trial={"cpu": 1},
                            num_samples=grid_size,
                            config=config,
                            stop={"training_iteration": 1},
                            reuse_actors=False)

        df = analysis.dataframe()
        print df[['config/a', 'config/b']]
```
output:
```
   config/a  config/b
0  0.654906  0.978321
1  0.404444  0.173096
2  0.448516  0.879822
```

where we would expect config/a == config/b.

A simple fix could be to add to _resolve_lambda_vars():
```
            if path in resolved:
                continue
```
i.e,
```
        for path, fn in lambda_vars:
            if path in resolved:
                continue
            try:
                value = fn(_UnresolvedAccessGuard(spec))
            except RecursiveDependencyError as e:
                error = e
            except Exception:
                raise ValueError(
                    "Failed to evaluate expression: {}: {}".format(path, fn))
            else:
                _assign_value(spec, path, value)
                resolved[path] = value
        ...
```

**Comment by richardliaw** [comment]
Hey @RoeeEnbar, thanks for opening this issue! Can you actually push a fix and a test for this? Sorry for the delay.

**Comment by krfricke** [comment]
This has been fixed in our search space refactoring #10444.
